lib/msf/core/post/file.rb

```diff
@@ -638,6 +642,153 @@ def copy_file(src_file, dst_file)
+      icacls_groups_permissions_array[i][0].strip! # Strip any leading or trailing characters off of each of the group permissions found via the regex.
+
+      # Permissions can exist inside a () section as comma delimited entries. To solve this we first remove the ( and ) characters,
+      # then we split the list on the delimiter, aka the comma character, before finally appending the resulting array to the end of
+      # final_permissions_array.
+      final_permissions_array << icacls_groups_permissions_array[i][0].gsub(/[()]/, '').split(',')
+    end
+    final_permissions_array
+  end
+
+  #
+  # Find what groups the current user is a member of.
+  #
+  # @return [Array] Array of groups the current user is a member of.
+  def win_find_current_user_groups
+    cuser_groups = cmd_exec('whoami /groups').to_s
+    cuser_groups_array = cuser_groups.scan(/^(?:\w+ ?[\w\-]*+ ?[\w\-]*+\\?[\w\-]*+ ?[\w\-]*+ ?[\w\-]*+ ?)/)
+    cuser_groups_length = cuser_groups_array.length
+    for i in 0...cuser_groups_length
+      cuser_groups_array[i].strip! # Strip any leading or trailing characters off the groups captured by the regex.
+    end
+
+    # Remove erronious entries that don't belong here but are captured by the regex.
+    cuser_groups_array.delete('GROUP INFORMATION')
+    cuser_groups_array.delete('Group Name')
+    cuser_groups_array
+  end
+
+  #
+  # See if +path+ on the remote system is accessible in the manner specified by +perm+ by the current user.
+  #
+  # @param path [String] Remote path to check
+  # @param perms [Array] Array of permissions to check for
+  #
+  # @return [Boolean] true if +path+ is accessible to the current user via any of the manners listed in +perms+, otherwise false.
+  def check_win_path_permissions(path, perms)
+    # Next check if icacls exists as a command on the target system, and switch to calcs if not.
+    if command_exists?('icacls')
+      # Grab permissions of the path in question and extract relevant info using a regex.
+      # Also strip extra whitespace before comparison occurs.
+      icacls_output_raw = cmd_exec("icacls \"#{path}\"").to_s
+
+      # Perform the actual regex extraction to determine what groups are permitted to
+      # access the file and what permissions each of those groups has on the file.
+      # For reference this is what the output from running "icacls" on win32kfull.sys looks like:
+      #
+      # C:\Users>icacls C:\Windows\System32\win32kfull.sys
+      # C:\Windows\System32\win32kfull.sys NT SERVICE\TrustedInstaller:(F)
+      #                                    BUILTIN\Administrators:(RX)
+      #                                    NT AUTHORITY\SYSTEM:(RX)
+      #                                    BUILTIN\Users:(RX)
+      #                                    APPLICATION PACKAGE AUTHORITY\ALL APPLICATION PACKAGES:(RX)
+      #
+      # Successfully processed 1 files; Failed processing 0 files
+      #
+      # C:\Users>
```

Comment:
Make this part of rdoc for `parse_icacls_output`

Reply:
This should also account for inheritance values. 
```
(I) - permission is inherited

(OI) - Object inherit
(CI) - Container inherit
(IO) - Inherit only
(NP) - Do not propagate inherit
```

Reply:
Not sure why we would want to account for inheritance values? All we care about is what permissions are there not if they were inherited or not as far as I can tell, though open to hearing opinions on this :)

Reply:
Inherited permissions might give hints as to how you might use paths also if enumerating the filesystem and some files or folders remove inheritance that could indicate the path contains data of value.  This is just supposition, however I can see a couple ideas for post modules that could benefit from full parsing of all NTFS the permissions.

Reply:
Resolved RDoc issue in https://github.com/rapid7/metasploit-framework/pull/15987/commits/9aee8e2b4264b96da724aa2f2606851d0cfe52b5